### PR TITLE
Mqtt integration

### PR DESCRIPTION
**Pull request by adi-miller**
Added this for my personal use, since I'm using MQTT. Let me know if this is interesting, if so I'll add more info in README. I also want to add filters (so that the user is able to specific specific areas that are relevant, as opposed to get MQTT notifications for all areas.

### CODE REVIEW

File: alarmpy/alarmpy.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from datetime import datetime
 from pathlib import Path
 from time import sleep, time
+import paho.mqtt.client as mqtt
```
Comment (yuvadm) [inline alarmpy/alarmpy.py]: If this were to be an optional dependency, I'd wrap this in a try block that catches an `ImportError` and passes

File: alarmpy/alarmpy.py [hunk 2]
```diff
@@ -25,19 +26,29 @@ def __init__(
...
+        self.mqtt_server = mqtt_server
+        self.mqtt_port = mqtt_port
+        self.mqtt_topic = mqtt_topic
 
         self.current_alarms = []
         self.last_routine_output = 0
 
         self.session = self.init_session()
         self.labels = self.load_labels()
+        self.mqtt = mqtt.Client("alamrPyClient")
```
Comment (yuvadm) [inline alarmpy/alarmpy.py]: Is this hard-coded client name by design?

File: setup.py [hunk 3]
```diff
@@ -26,6 +26,6 @@
     packages=setuptools.find_packages(exclude="tests"),
     package_data={"alarmpy": ["data/*.json"]},
     python_requires=">=3.6",
-    install_requires=["requests", "click"],
+    install_requires=["requests", "click", "paho.mqtt.client"],
```
Comment (yuvadm) [inline setup.py]: Should probably be an optional dependency

**Review by yuvadm — CHANGES_REQUESTED** [review]
Interesting feature! However, I would probably add MQTT support, with the additional dependency it requires, as optional - see https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#optional-dependencies

File: alarmpy/alarmpy.py (hunk 2 above)
Comment (adi-miller) [reply alarmpy/alarmpy.py]: I didn't want to clutter the config. Will fix.

File: setup.py (hunk 3 above)
Comment (adi-miller) [reply setup.py]: Not sure how to do that. Can you please advise?

Comment (yuvadm) [reply setup.py]: Check out https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#optional-dependencies

**Review by yuvadm — APPROVED** [review]
Looks good! I'll merge this into `mqtt` and do some testing and cleanup, but otherwise this is a great feature, thanks @adi-miller :+1:

**Pull request merged**